.github/workflows/build_and_test.yml

```diff
@@ -300,16 +300,27 @@ jobs:
         uses: docker/setup-qemu-action@v1
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
-      - name: Build and push
+      - name: Build and push (Infra)
         id: docker_build
         uses: docker/build-push-action@v2
         with:
-          context: ./dev/infra/
+          file: ./dev/infra/Dockerfile
           push: true
           tags: |
             ${{ needs.precondition.outputs.image_url }}
           # Use the infra image cache to speed up
           cache-from: type=registry,ref=ghcr.io/apache/spark/apache-spark-github-action-image-cache:${{ inputs.branch }}
+      - name: Build and push (Lint)
+        id: lint_docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: ./dev/infra/lint.Dockerfile
+          push: true
+          tags: |
+            ${{ needs.precondition.outputs.image_url }}-lint
+          build-args: BASE_IMG=ghcr.io/apache/spark/apache-spark-github-action-image-cache:${{ inputs.branch }}-static
```

Comment:
This might bring some complex and issue when first commit merge after cut down branch because base image haven't generated yet. But all in one infra image doesn't have this problem, becuase only cache not working in first commit after cut down branch.